### PR TITLE
Tests/Twig: POC with initial test

### PR DESCRIPTION
**Pull request by justintemps**
Sets up tests with Jest and provides a single test as POC.

To try it out:

```
$ pnpm --filter twig test
```

### CODE REVIEW

**Comment by bashlk** [comment]
@justintemps The solution proposed is really elegant but I have some concerns

- twig-testing-library is not actively maintained and hasn't reached sufficient adoption to have a lot of community contributions.
- twig-testing-library uses twig.js to render the twig templates and from what I can see, it is a completely separate implementation of twig which openly states that it is "a work in progress and supports a limited subset of the Twig templating language". I imagine that it should have sufficient support for the twig constructs that we are using but there can always be discrepancies with how twig.js renders a templates vs how the original twig engine renders it on php. (the creators of the twig-testing-library have already run into some issues around includes for example https://github.com/larowlan/twig-testing-library/pull/11)

I think it might be better if we go a step higher and test within the browser. What I mean by this is to have a kind of testing server that renders the twig templates, ideally using drupal and then write tests that assert on the rendered component within the browser (using either playwright or cypress - for example). Another benefit of this is that the tests can be reused for both the twig and react versions of the components and the tests will test the components as a user would see it, using the same / similar stack as a real site, rather than testing implementation details in a testing specific environment.

I'm imagining a simple server where you make requests with the component props (e.g. twig-test-server.local/card?size=narrow&cornercut=true). I imagine that once set up, it would only need to be expanded to handle more components - the props can always be parsed from the URL and set.

File: packages/twig/setup-test-env.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import "@testing-library/jest-dom/extend-expect";
+
+global.Drupal = {
+  behaviors: {},
+};
+
+global.prefix = "ilo";
```
[the remark this answers is not in the document]
Comment (justintemps) [reply packages/twig/setup-test-env.js:7]: @bashlk Future versions of the Design System will be "themable" so if you need a different namespace than ilo for classnames, then you can get it through configuration.

File: packages/twig/package.json [hunk 2]
```diff
@@ -89,6 +92,7 @@
     "storybook-addon-rtl-direction": "^0.0.19",
     "style-loader": "^1.3.0",
     "stylelint-scss": "^3.21.0",
+    "twig-testing-library": "^1.2.0",
```
[the remark this answers is not in the document]
Comment (justintemps) [reply packages/twig/package.json:95]: @bashlk this is true

**Comment by justintemps** [comment]
@bashlk You're right this definitely has limitations. 

It seems like what you're describing could be achieved through a Github action workflow? On PR or similar event, it would:

1. Checkout the branch in question
2. Launch the testing server in a docker container
3. Run tests against components rendered by the testing server

That could include snapshot tests etc. that would do what you're saying (e.g. test what components do as opposed to how they're implemented).

However, because this is a library and not a website, I think we also have to test the implementation to detect when we're introducing breaking changes without realising it.

For example, changing the name of a token in the `themes` package is a breaking change, a unit test would flag that whereas an E-to-E test might not.

**Comment by bashlk** [comment]
@justintemps Yes exactly, we can have this run within Github actions and also have scripts to start the server locally for test development. The test server should attempt to use the components as close to how they would be used by a real consumer.

Just a note that in my experience, snapshot tests are almost never maintained in the long term since they are too sensitive and easy to bork since they can be updated to pass even if the component is not working as intended. Specific tests that assert specific behaviour of the component are the ones that are easier to maintain and the most useful imo.

I think we can have the e2e tests halt when there are breaking changes by having the build process fail when it fails to import a value from a separate package. In general, I think we can get the e2e tests to test for breaking changes as well without explicitly testing implementation details.

**Pull request closed without merging**